clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -521,8 +523,10 @@ private int select(long ms) throws IOException {
      */
     public void close(String id) {
         KafkaChannel channel = this.channels.get(id);
-        if (channel != null)
+        if (channel != null) {
+            channel.state(ChannelState.LOCAL_CLOSE);
             close(channel, false);
```

Comment:
By passing `false` here, the channel is never added to the `disconnected` list and since we remove the channel from `channels`, it seems like this state transition is never observed. Is that correct?